pypeit/core/coadd.py

```diff
@@ -766,19 +799,36 @@ def sn_weights(waves, fluxes, ivars, masks, sn_smooth_npix, const_weights=False,
             weights[:, iexp] = sn_conv
     else:
         weights = np.zeros_like(flux_stack)
+        refscale = (sn_val[:, ref_spec] > 0) / (sn_val[:, ref_spec] + (sn_val[:, ref_spec] == 0))
         for iexp in range(nstack):
+            if relative_weights:
+                if iexp == ref_spec:
+                    # Reference spectrum gets a weight of 1
+                    weights[:, iexp] = 1.0
+                    continue
+                else:
+                    # Compute the relative (S/N)^2 and update the mask
+                    sn2[iexp] /= sn2[ref_spec]
+                    mask_stack[:, iexp] *= (mask_stack[:, ref_spec]) | (sn_val[:, ref_spec] != 0)
+                    sn_val[:, iexp] *= refscale
```

Comment:
You are dividing a noisy pixel based estimate of the S/N ratio by another noisy (although less noisy since it is the reference spectrum) pixel based estimate of the S/N ratio. I fail to see why dividing the ratio of two noisy quantities would ever produce better results than the quantities themselves.